### PR TITLE
Minor improvements for  CardTable2 model

### PR DESCRIPTION
**Pull request by magicDGS**
- Move jackson `JsonSerialize` annotation to the model itself to allow list mappings in an easy way
- Split by new line the set fields to simplfiy further processing
- Fix a possibel NPE for the `MarkupStringMapper`
- Add a way to type integer values that might be undefined (used for atk/def)
- Type link arrows as a list of strings
- Add missing `pt_name` to `CardTable2` model

### CODE REVIEW

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/model/IntegerOrUndefined.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package io.github.ygojson.tools.dataprovider.impl.yugipedia.model;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
+
+/**
+ * Identifies an integer that can be undefined by matching '?'.
+ */
+@JsonSerialize(using = ToStringSerializer.class)
+public class IntegerOrUndefined {
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/model/IntegerOrUndefined.java:10]: Add tests to define the behavior that we want with different kind of values and adapt as expected.

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/MarkupStringMapper.java [hunk 2]
```diff
@@ -43,6 +45,18 @@ public MarkupString mapToJapanese(final String property) {
...
+		}
 		return property.withoutMarkup();
 	}
+
+	public List<MarkupString> mapToNewLineList(final String property) {
+		return map(property).splitByNewLine(true).toList();
+	}
+
+	@Named("commaSeparatedList")
+	public List<String> mapToCommaSeparatedList(final String property) {
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/MarkupStringMapper.java:59]:
```suggestion
	public List<String> mapToCommaSeparatedStringList(final String property) {
```

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/MarkupStringMapper.java [hunk 3]
```diff
@@ -43,6 +45,18 @@ public MarkupString mapToJapanese(final String property) {
...
+			return null;
+		}
 		return property.withoutMarkup();
 	}
+
+	public List<MarkupString> mapToNewLineList(final String property) {
+		return map(property).splitByNewLine(true).toList();
+	}
+
+	@Named("commaSeparatedList")
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/MarkupStringMapper.java:58]:
```suggestion
	@Named("commaSeparatedStringList")
```

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/MarkupStringMapper.java [hunk 4]
```diff
@@ -43,6 +45,18 @@ public MarkupString mapToJapanese(final String property) {
...
 	 */
 	public String map(final MarkupString property) {
+		if (property == null) {
+			return null;
+		}
 		return property.withoutMarkup();
 	}
+
+	public List<MarkupString> mapToNewLineList(final String property) {
+		return map(property).splitByNewLine(true).toList();
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/MarkupStringMapper.java:55]: Add some tests to the `MarkupStringMapper` new methods to check for the null values. This will throw a NPE and we want to prevent it, so also adapt

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java [hunk 5]
```diff
@@ -90,6 +90,7 @@ private Map<String, String> wikitextToMap(final String wikitext) {
 	@Mapping(target = "ja_name", qualifiedByName = "japanese")
 	@Mapping(target = "ja_pendulum_effect", qualifiedByName = "japanese")
 	@Mapping(target = "ja_lore", qualifiedByName = "japanese")
+	@Mapping(target = "link_arrows", qualifiedByName = "commaSeparatedList")
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java:93]:
```suggestion
	@Mapping(target = "link_arrows", qualifiedByName = "commaSeparatedStringList")
```

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/MarkupStringMapper.java (hunk 4 above)
Comment (magicDGS) [reply tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/MarkupStringMapper.java:55]: This doesn't throw a NPE because the MarkupString container accept null values and handle them propertly. Anyway, tests were implemented.

**Review by magicDGS — COMMENTED** [review]
LGTM
